src/index.js

```diff
@@ -1,65 +1,9 @@
-/**
- * WordPress dependencies
- */
-import { addAction, addFilter } from '@wordpress/hooks';
-import {
-	doGutenbergNativeSetup,
-	initialHtmlGutenberg,
-} from '@wordpress/react-native-editor';
-
 /**
  * Internal dependencies
  */
-import correctTextFontWeight from './text-font-weight-correct';
-import {
-	registerJetpackBlocks,
-	registerJetpackEmbedVariations,
-	setupJetpackEditor,
-} from './jetpack-editor-setup';
-import setupBlockExperiments from './block-experiments-setup';
-import initialHtml from './initial-html';
-import initAnalytics from './analytics';
-
-addAction( 'native.pre-render', 'gutenberg-mobile', ( props ) => {
-	require( './strings-overrides' );
-	correctTextFontWeight();
-
-	setupJetpackEditor(
-		props.jetpackState || { blogId: 1, isJetpackActive: true }
-	);
-
-	// Jetpack Embed variations use WP hooks that are attached to
-	// block type registration, so it’s required to add them before
-	// the core blocks are registered.
-	registerJetpackEmbedVariations( props );
-} );
+import { registerGutenberg } from '@wordpress/react-native-editor';
 
-addAction( 'native.render', 'gutenberg-mobile', ( props ) => {
-	const capabilities = props.capabilities ?? {};
-	registerJetpackBlocks( props );
-	setupBlockExperiments( capabilities );
+registerGutenberg( () => {
+	const { default: gutenbergMobileSetup } = require( './setup' );
```

Comment:
The setup code is lazy-loaded to guarantee that we can execute code, like setting the locale data, before any other code is imported or initialized. 

**NOTE:** Although this is not required yet, it will be needed when adding the translations of plugins like Jetpack or Layout Grid.